Sources/Components/CarouselView/CarouselView.swift

```diff
@@ -0,0 +1,192 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+public typealias CarouselViewCell = UICollectionViewCell
+
+public protocol CarouselViewDataSource: AnyObject {
+    func numberOfItems(in carouselView: CarouselView) -> Int
+    func carouselView(_ carouselView: CarouselView, cellForItemAt indexPath: IndexPath) -> CarouselViewCell
+}
+
+public protocol CarouselViewDelegate: AnyObject {
+    func carouselView(_ carouselView: CarouselView, didSelectItemAt indexPath: IndexPath)
+    func carouselViewDidEndDecelerating(_ carouselView: CarouselView)
+}
+
+public class CarouselView: UIView {
+
+    public weak var dataSource: CarouselViewDataSource? {
+        didSet {
+            collectionView.reloadData()
+        }
+    }
+
+    public weak var delegate: CarouselViewDelegate?
+
+    // MARK: - Private Properties
+
+    private var initialLayout = true
```

Comment:
Should we change this to more "bool-friendly" name, like for example `shouldSetInitialLayout` or something like that?

Reply:
I like it!